api/src/test/java/jakarta/data/BasicRestrictionRecordTest.java

```diff
@@ -62,6 +65,23 @@ void shouldCreateBasicRestrictionWithNullValue() {
         });
     }
 
+    @Test
+    void shouldNegateLTERestriction() {
```

Comment:
I should have a test scenario case when we negate twice, I mean negate a Restriction that is already `not`

Reply:
I mean, not remove this one, but include a second one

Reply:
> I should have a test scenario case when we negate twice, I mean negate a Restriction that is already `not`

Good idea. I added tests for negating a negated restriction here:
https://github.com/jakartaee/data/pull/905/commits/f5d3bb91143dd534ab0cda7089cf80b906a9dbc7